src/jit/rationalize.cpp

```diff
@@ -93,14 +93,33 @@ void Rationalizer::RewriteSIMDOperand(LIR::Use& use, bool keepBlk)
         return;
     }
 
-    // If we have GT_IND(GT_LCL_VAR_ADDR) and the GT_LCL_VAR_ADDR is TYP_BYREF/TYP_I_IMPL,
-    // and the var is a SIMD type, replace the expression by GT_LCL_VAR.
+    // If we have GT_IND(GT_LCL_VAR_ADDR) and the var is a SIMD type,
+    // replace the expression by GT_LCL_VAR or GT_LCL_FLD.
     GenTree* addr = tree->AsIndir()->Addr();
-    if (addr->OperIsLocalAddr() && comp->isAddrOfSIMDType(addr))
+    if (addr->OperIs(GT_LCL_VAR_ADDR) && comp->lvaGetDesc(addr->AsLclVar())->lvSIMDType)
     {
         BlockRange().Remove(tree);
 
-        addr->SetOper(loadForm(addr->OperGet()));
+        var_types lclType = comp->lvaGetDesc(addr->AsLclVar())->TypeGet();
+
+        if (lclType == simdType)
+        {
+            addr->SetOper(GT_LCL_VAR);
+        }
+        else
+        {
+            addr->SetOper(GT_LCL_FLD);
+            addr->AsLclFld()->gtLclOffs  = 0;
+            addr->AsLclFld()->gtFieldSeq = FieldSeqStore::NotAField();
+
+            if (((addr->gtFlags & GTF_VAR_DEF) != 0) && (genTypeSize(simdType) < genTypeSize(lclType)))
```

Comment:
Could we see `genTypeSize(simdType) > genTypeSize(lclType)` here? Probably not because you can't do `Unsafe.As<Vector3, Vector4>(ref c);` , but if yes then you can use `bool GenTree::IsPartialLclFld(Compiler* comp)` here.

Reply:
I assume it's possible, but the result will be undefined. If it's a local variable you may end up overwriting another local variable but that's not something that liveness would ever care about.

The `USEASG` case is special - if this isn't set correctly, liveness will believe that the entire variable was written to and it will eliminate any previous stores as dead.

BTW, I'm not sure if this particular case is ever hit. It seems technically possible to hit a `VAR_DEF` but I haven't been able to construct a case showing this so I only added this code out of caution. I hope that eventually we will get rid of the entire function, all this logic should be either in morph or in lowering.